### PR TITLE
Message History - 2/N

### PR DESCRIPTION
**Pull request by tuddman**
## Summary

More work towards supporting Message History.

This PR includes 
- `proto` changes 
- stubs for handling message history messages.
- `sync` group creation function
- a separate filtered `find_sync_groups` method to only find the `sync` group types.
- adds initital tests

### CODE REVIEW

File: xmtp_mls/src/storage/encrypted_store/group.rs [hunk 1]
```diff
@@ -82,6 +99,14 @@ impl DbConnection<'_> {
         Ok(self.raw_query(|conn| query.load(conn))?)
     }
 
+    /// Return only the [`Purpose::Sync`] groups
+    pub fn find_sync_groups(&self) -> Result<Vec<StoredGroup>, StorageError> {
```
Comment (neekolas) [inline xmtp_mls/src/storage/encrypted_store/group.rs:103]: I guess there isn't really a reason to ever look for all sync and conversation groups at the same time, right?

**Review by neekolas — APPROVED** [review]

File: xmtp_mls/src/groups/sync.rs [hunk 2]
```diff
@@ -262,38 +267,38 @@ where
...
+                            sender_installation_id,
+                            sender_account_address,
+                            delivery_status: DeliveryStatus::Published,
+                        }
+                        .store(provider.conn())?
+                    }
+                    Some(Content::V2(_)) => {
+                        todo!()
+                    }
+                    None => return Err(MessageProcessingError::InvalidPayload),
```
Comment (richardhuaaa) [inline xmtp_mls/src/groups/sync.rs:300]: Double-checking, are the changes in this file purely refactoring, or is there a functional effect too?

File: xmtp_mls/src/storage/encrypted_store/group.rs [hunk 3]
```diff
@@ -49,9 +50,25 @@ impl StoredGroup {
...
+            membership_state,
+            installations_last_checked: 0,
+            purpose: Purpose::Sync,
+        }
+    }
 }
 
 impl DbConnection<'_> {
+    /// Return  regular [`Purpose::Conversation`] groups with additional filters
     pub fn find_groups(
```
Comment (richardhuaaa) [inline xmtp_mls/src/storage/encrypted_store/group.rs:72]: We need to add the code change to filter out the sync groups here (or filter to Conversation groups) right?

**Review by richardhuaaa — APPROVED** [review]

File: xmtp_mls/src/storage/encrypted_store/group.rs (hunk 1 above)
Comment (tuddman) [reply xmtp_mls/src/storage/encrypted_store/group.rs:103]: right, AFAICT.  I couldn't think of a scnerio where we'd want to. 
And we explicitly want to keep `listGroups()` as it is, for only returning "normal (conversation)" groups.

If we should find a need later to look for all groups at the same time, we'll add it if/when that need arises.

File: xmtp_mls/src/groups/sync.rs (hunk 2 above)
Comment (tuddman) [reply xmtp_mls/src/groups/sync.rs:300]: you're correct - it is purely a refactoring (and stubbing support for the `V2` match arm).
I saw that I was needlessly pulling out the `key` and `message` just to use a few lines later.   That is now self-contained within the match arm.

File: xmtp_mls/src/storage/encrypted_store/group.rs (hunk 3 above)
Comment (tuddman) [reply xmtp_mls/src/storage/encrypted_store/group.rs:72]: yep, that's already happening from my [last PR](https://github.com/xmtp/libxmtp/pull/553/files#diff-a7388e26da76ba3ef25a9251cfca5a9b66a99e7d91e3d041c6dce54e4618feeaR80) 😉

**Pull request merged**